.github/workflows/build.yml

```diff
@@ -97,36 +84,36 @@ jobs:
           echo ::set-output name=extension::gz
         fi
 
-    - name: (not check) Upload server to release
-      if: ${{ !contains(github.ref_name, 'check') }}
+    - name: Upload server to release
+      if: ${{ !contains(github.ref_name, 'check') && github.event.release.upload_url != ''}}
       uses: actions/upload-release-asset@v1.0.2
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         upload_url: ${{ github.event.release.upload_url }}
         asset_path: ${{ steps.compress_server_binary.outputs.path }}
-        asset_name: haskell-language-server-${{ runner.OS }}-${{ env.GHC_VERSION }}${{env.EXE_EXT}}.${{ steps.compress_server_binary.outputs.extension }}
+        asset_name: haskell-language-server-${{ runner.os }}-${{ matrix.ghc  }}${{env.EXE_EXT}}.${{ steps.compress_server_binary.outputs.extension }}
         asset_content_type: ${{ steps.compress_server_binary.outputs.content_type }}
 
     - name: Upload server to workflow artifacts
       uses: actions/upload-artifact@v2
       with:
-        name: haskell-language-server-${{ runner.OS }}-${{ matrix.ghc }}${{env.EXE_EXT}}.${{ steps.compress_server_binary.outputs.extension }}
+        name: haskell-language-server-${{ runner.os }}-${{ matrix.ghc }}${{ env.EXE_EXT }}.${{ steps.compress_server_binary.outputs.extension }}
         path: ${{ steps.compress_server_binary.outputs.path }}
 
-    - name: (GHC 8.10) Build the wrapper
+    - name: Build the wrapper
       if: matrix.ghc == '8.10.7'
       run: cabal build exe:hls-wrapper -O2 $LINUX_CABAL_ARGS
```

Comment:
does this need to be `$CABAL_ARGS` ?

Reply:
omg 😧

Reply:
that should have failed, as @hasufell suggested, setting the appropriate bash flag